src/osd/OSD.cc

```diff
@@ -2205,7 +2205,18 @@ bool OSD::asok_command(string admin_command, cmdmap_t& cmdmap, string format,
       pg->unlock();
     }
     f->close_section();
-  } else {
+  } else if (admin_command == "compact") {
+    dout(1) << "triggering manual compaction" << dendl;
+    utime_t start = ceph_clock_now();
+    store->compact();
```

Comment:
Is this a synchronous operation or async? If it is a sync operation, what is the impact to the front end IO's? If there is some impact, then the users should know about it as warning. Did you try this a live osd when writes are in progress?